Restore last vmd session's window state on startup

supercrabtree opened the pull request:
Saves the state of the vmd window on save and then restores it when vmd is opened again.

Is a feature I wanted as the first thing I always do when calling `vmd` is move the window.

maxkueng commented:
Hey George! That's an awesome feature :100: :bangbang: 

I tried it out and I noticed that the `electron-window-state` module stores the window state in a JSON file in `path.join(app.getPath('userData'), 'window-state.json')` (see [electron-window-state:index.js#L23](https://github.com/mawie81/electron-window-state/blob/master/index.js#L23)) which turns out to be `/home/max/.config/Electron/window-state.json`. On Linux at least, probably similar on OS X. This means that it will clash with other Electron apps that use the same module and don't use a custom path. And I don't think Electron's config directory is the correct place to store this information.

vmd reads its own configs from the following locations: `$HOME/.vmdrc`, `$HOME/.vmd/config`, `$HOME/.config/vmd`, `$HOME/.config/vmd/config`, `/etc/vmdrc`, and [a number of other places](https://github.com/maxkueng/rucola#configuration-sources). So these can't be used either even if the window state was merged with existing settings as the format does not necessarily have to be JSON (vmd supports JSON, INI and YAML).

The easiest solution would probably be to set the `file` option in `windowStateKeeper()` to something like `"vmd-window-state.json"` but then the file would still be in Electron's config directory which is the wrong place IMO. Or maybe Electron can be configured so that `app.getPath('userData')` returns something app-specific.

Another option would be to store the file alongside vmd's config files but vmd may not always have it's own directory (e.g. `~/.vmdrc` is not a directory). But It's not ideal either since this type of state is more _cache_ rather than _configuration_.

The third option would be to handle window state preservation in the renderer process and store the state in `localStorage`. Then the information would be somewhere in `~/.cache` (or similar) which is where one would expect persisted application state.

A killer bonus feature would be if window state preservation could be turned off using a configuration option. For example `vmd --window-preservestate=false` and `window.preserverstate = false` in the config file. Some people have external programs that manage the positioning of windows and this could interfere with them. It could be turned on by default.

supercrabtree commented:
Hi, I had a look into the localStorage option, but it seemed pretty involved and I don't really have heaps of time at the moment 😔️ 

I changed the `file` option in `windowStateKeeper()` to "vmd-window-state.json" and added the `window.preservestate = true` default.

Hope this is all good, let me know if you want me to squash.

maxkueng commented:
Yeah I guess that's best for now to get the feature in. Maybe it can be refactored later. I'll publish a new release in couple secs.

Thanks!

maxkueng commented:
vmd@1.27.0 is on npm :tada:
